modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySpi.java

```diff
@@ -686,7 +688,7 @@ public TcpDiscoverySpi setLocalPort(int locPort) {
      */
     @IgniteSpiConfiguration(optional = true)
     public TcpDiscoverySpi setLocalPortRange(int locPortRange) {
-        this.locPortRange = locPortRange;
+        this.locPortRange = locPortRange == 0 ? 1 : locPortRange;
```

Comment:
The same as above. Please don't set locPortRange to 1 here because getLocalPortRange() method must return '0' if the latest is set.

Modify loops in TcpCommunicationSpi the same way as I proposed in ServerImpl above.
Look for usages of TcpCommunicationSpi.locPortRange and you will find a couple of places where locPortRange is used.